spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/Tracer.java

```diff
@@ -72,25 +82,33 @@
 	/**
 	 * Start a new span if the sampler allows it or if we are already tracing in this
 	 * thread. A sampler can be used to limit the number of traces created.
-	 *  @param name the name of the span
+	 *
+	 * @param name the name of the span
 	 * @param sampler a sampler to decide whether to create the span or not
 	 */
 	Span startTrace(String name, Sampler sampler);
 
 	/**
-	 * Pick up an existing span from another thread.
+	 * Pick up an existing span from another thread. Continuing a span signifies that
+	 * all changes applied to the returned span from this method will be also
+	 * applied to the continued span (the one passed as the argument).
 	 */
 	Span continueSpan(Span span);
 
 	/**
-	 * Adds a tag to the current span if tracing is currently on.
+	 * <p>Adds a tag to the current span if tracing is currently on.</p>
+	 *
+	 * <p>
+	 *     Every span may also have zero or more key/value Tags, which do not have
+	 *     timestamps and simply annotate the spans.
+	 * </p>
 	 */
 	void addTag(String key, String value);
 
 	/**
 	 * Remove this span from the current thread, but don't stop it yet or send it for
 	 * collection. This is useful if the span object is then passed to another thread for
```

Comment:
this is useful because.. :)

My guess is that this allows you "assign ownership" of the span, which eliminates the shared mutable state problem. Maybe an example?  Ex. nextTracer.continueSpan(currentTracer.detach(span))?

forged example, but basically this looks advanced so we should reinforce what it's for.